backend/common/services/cache_service.py

```
@@ -270,6 +273,26 @@ def scard(self, key: str) -> int:
+        except RedisError as e:
+            logger.error(f"[REDIS] Error smembers for {key}: {e}")
+            return set()
+    
+    def keys(self, pattern: str) -> List[str]:
+        """Get all keys matching a pattern. Use sparingly - can be slow on large datasets."""
+        if not self._ensure_connected():
+            return []
+        try:
+            return self.redis.keys(pattern)
+        except RedisError as e:
+            logger.error(f"[REDIS] Error keys for pattern {pattern}: {e}")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)